smt/surrogate_models/krg_based.py

```diff
@@ -1647,6 +1649,8 @@ def _predict_variances(self, x: np.ndarray, is_acting=None) -> np.ndarray:
             X_cont, _ = compute_X_cont(x, self.design_space)
             X_cont = (X_cont - self.X_offset) / self.X_scale
         else:
+            n_eval, n_features_x = x.shape
+            _, ij = cross_distances(x, self.X_train)
```

Comment:
Same comments as above. Actually, you should create a function to remove code duplication.

Reply:
Not solved for me. Could you extract common init code between `_predict_values` and `_predict_variances` in a function `_predict_init()`? In order to replace code duplication by:
```
r, X_cont = self._predict_init(self, ...)
```